dask/array/core.py

```diff
@@ -1109,12 +1109,20 @@ def itemsize(self):
         """ Length of one array element in bytes """
         return self.dtype.itemsize
 
+    @property
+    def name(self):
+        return self._name
+
+    @name.setter
+    def name(self, val):
+        self._name = val
+        # Clear the key cache when the name is reset
+        self._cached_keys = None
```

Comment:
I'd rather keep things localized for now, we can always break things out later as needed.